pkg/auth/providers/publicapi/login.go

```diff
@@ -130,9 +138,17 @@ func (h *loginHandler) createLoginToken(request *types.APIContext) (v3.Token, st
 	}
 
 	// Authenticate User
-	userPrincipal, groupPrincipals, providerToken, err := providers.AuthenticateUser(input, providerName)
-	if err != nil {
-		return v3.Token{}, "", err
+	if providerName != saml.PingName {
```

Comment:
I want a more clearly defined boundary between saml and non-saml flow. THis doesnt make it obvious why saml.PingName is special. Can you rearrrange to something like:
```
// Some comment exlpaining that saml has a different flow that requires us to short circuit the normal flow
if providerName == saml.Ping {
   ...
}

userPrincipal, groupPrincipals, providerToken, err = providers.AuthenticateUser(input, providerName)
```